iota/transaction/base.py

```diff
@@ -98,94 +205,165 @@ def __init__(
     ):
         self.hash = hash_
         """
-        Transaction ID, generated by taking a hash of the transaction
-        trits.
+        The transaction hash, used to uniquely identify the transaction on the
+        Tangle.
+
+        This value is generated by taking a hash of the raw transaction trits.
+
+        :type: :py:class:`TransactionHash`
         """
 
         self.bundle_hash = bundle_hash
         """
-        Bundle hash, generated by taking a hash of metadata from all the
+        The bundle hash, used to identify transactions that are part of the same
+        bundle.
+
+        This value is generated by taking a hash of the metadata from all
         transactions in the bundle.
+
+        :type: :py:class:`BundleHash`
         """
 
         self.address = address
         """
         The address associated with this transaction.
 
-        If ``value`` is != 0, the associated address' balance is
+        Depending on the transaction's ``value``, this address may be a sender
+        or a recipient. If ``value`` is != 0, the associated address' balance is
         adjusted as a result of this transaction.
+
+        :type: :py:class:`Address`
         """
 
         self.value = value
         """
-        Amount to adjust the balance of ``address``.
-        Can be negative (i.e., for spending inputs).
+        The number of IOTAs being transferred in this transaction:
```

Comment:
```suggestion
        The number of iotas being transferred in this transaction:
```

(to make consistent with other parts of this PR)